### DISCUSSION TITLE
[Feature Proposal] Add a function with an api similar to styled-system that returns tailwind classes

### DISCUSSION THREAD

**Discussion by jeffreyyoung** [tailwindlabs/tailwindcss#1149] [Ideas]
I think it would be very useful to have a function that takes args in the same shape as [@styled-system/css](https://styled-system.com/css) and generates tailwind classes.

The api would be something like this
```javascript
import css from 'tailwind/css';

let myClass = css({
     fontSize: [ "sm", "base", "lg" ],
    color: 'primary',
    bg: 'gray',
    '&:hover': {
      color: 'secondary',
    },
});
console.log(myClass); // "text-sm md:text-base lg:text-lg text-primary bg-gray hover:text-secondary"
<div class={myClass}></div>
```

Ideally this function would include typescript types generated from the current theme to help developers recognize invalid arguments.

**Comment by hacknug** [comment]
Have you seen this project? https://github.com/bradlc/babel-plugin-tailwind-components/
